packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1676,29 +1676,41 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
   void _updateScrollPosition(Offset updatedOffset) {
     assert(_cachedController != null);
     assert(_startDragScrollbarAxisOffset != null);
+    assert(_lastDragUpdateOffset != null);
     assert(_startDragThumbOffset != null);
 
     final ScrollPosition position = _cachedController!.position;
-    late double primaryDelta;
+    late double primaryDeltaFromDragStart;
+    late double primaryDeltaFromLastDragUpdate;
     switch (position.axisDirection) {
       case AxisDirection.up:
-        primaryDelta = _startDragScrollbarAxisOffset!.dy - updatedOffset.dy;
+        primaryDeltaFromDragStart = _startDragScrollbarAxisOffset!.dy - updatedOffset.dy;
+        primaryDeltaFromLastDragUpdate = _lastDragUpdateOffset!.dy - updatedOffset.dy;
         break;
       case AxisDirection.right:
-        primaryDelta = updatedOffset.dx -_startDragScrollbarAxisOffset!.dx;
+        primaryDeltaFromDragStart = updatedOffset.dx -_startDragScrollbarAxisOffset!.dx;
+        primaryDeltaFromLastDragUpdate = updatedOffset.dx -_lastDragUpdateOffset!.dx;
         break;
       case AxisDirection.down:
-        primaryDelta = updatedOffset.dy -_startDragScrollbarAxisOffset!.dy;
+        primaryDeltaFromDragStart = updatedOffset.dy -_startDragScrollbarAxisOffset!.dy;
+        primaryDeltaFromLastDragUpdate = updatedOffset.dy -_lastDragUpdateOffset!.dy;
         break;
       case AxisDirection.left:
-        primaryDelta = _startDragScrollbarAxisOffset!.dx - updatedOffset.dx;
+        primaryDeltaFromDragStart = _startDragScrollbarAxisOffset!.dx - updatedOffset.dx;
+        primaryDeltaFromLastDragUpdate = _lastDragUpdateOffset!.dx - updatedOffset.dx;
         break;
     }
 
     // Convert primaryDelta, the amount that the scrollbar moved since the last
-    // time when drag started, into the coordinate space of the scroll
+    // time when drag started or last updated, into the coordinate space of the scroll
     // position, and jump to that position.
-    final double scrollOffsetGlobal = scrollbarPainter.getTrackToScroll(primaryDelta + _startDragThumbOffset!);
+    double scrollOffsetGlobal = scrollbarPainter.getTrackToScroll(primaryDeltaFromDragStart + _startDragThumbOffset!);
```

Comment:
When the scroll metrics shrink, fall back to the previous logic(before #112434) and give up making the scrollbar follow the pointer. However, the stuttering of the bar is unavoidable as the metrics get bigger and smaller.

Reply:
In this case, why would the scroll metrics shrink? The ListView's children don't change.

Reply:
If I remember correctly, in the lazy loading list, it is estimated according to the size of the current item.

Reply:
Interesting. That doesn't seem like a great heuristic in this case. Allowing the max scroll extents to shrink implies that we're guessing that items that have been scrolled out of view are smaller because the current (in view) items are smaller. That could certainly happen, but it doesn't seem like a common case when the list's children aren't changing.

Reply:
I have checked a few native apps and seen that @xu-baolin's behavior here is the most common. The scrollbar will instead jump away from the pointer, but remain engaged with the drag gesture, allowing the pointer and the bar to continue separate from each other.
My first thought was to keep the thumb with the pointer and then update it after the drag is released, but that is not the behavior I am seeing on multiple apps on mobile and desktop. 
Developers may want to be able to choose between the two, but I think for right now we should match the native behavior and maybe file an issue to see if there is interest in the other.